solutions/silver/cf-1879D.mdx

```diff
@@ -0,0 +1,78 @@
+---
+id: cf-1879D
+source: CF 1879D
+title: Sum of XOR Functions
+author: Justin Ji
+---
+
+<Warning title="Be careful with integer overflow!">
+
+Because we are working with very large numbers, the problem statement asks us to
+take the answer by some very large modulo. For more details on how to handle modular
+arithmetic, you can read the [Modular Arithmetic](https://usaco.guide/gold/modular)
+on this guide.
+
+</Warning>
+
+[Official Analysis (C++)](https://codeforces.com/blog/entry/120773)
+
+## Explanation
+
+Let's try to simplify the problem. Consider processing the array bit by bit.
+Each bit $x$ contributes $2^x \cdot \sum_{l=1}^{n} \sum_{r=l}^{n} p \cdot
+(r - l + 1)$ to the answer, where $p$ is the parity of this bit's occurrences.
+Because of this, it becomes apparent that we need to find the sum of subarray
+lengths for all subarrays that have this bit show up an odd amount of times.
+
+For an index $i$, we want to find the sum of subarray lengths for all relevant
+subarrays that end with $i$. This sum can be expressed as
+$ \sum_{l=1}^{x} i-v[l]+1$, where $v$ is the set of all valid left
+endpoints, and $x$ is the number of such endpoints. This expression is easily
```

Comment:
stop using easily pls
we are a wholesome organization

Reply:
^^^ thanks for catching that